File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        nsum = 1
        count = 0
        for i in nums:
            nsum = i * nsum
            if i == 0:
                count+=1
        ans = []
        tmp = 1
        for i in nums:
            if count >=2:
                ans.append(0)    
            elif i!=0:
                ans.append(nsum//i)
            elif i==0:
                for j in nums:
                    if j==0:
                        continue
                    else:
                        tmp = tmp*j
                ans.append(tmp) 
        return ans
            

File: Data Structures & Algorithms/permutations/submission-0.cpp
class Solution {
public:
    vector<vector<int>> ans;
    void dfs(vector<int>& nums, int i){
        if (i == nums.size()){
            ans.push_back(nums);
            return;
        }
        for (int j = i;j < nums.size(); j++){
            swap(nums[i],nums[j]);
            dfs(nums,i+1);
            swap(nums[i],nums[j]);
        }
    }
    vector<vector<int>> permute(vector<int>& nums) {
        dfs(nums,0);
        return ans;
    }
};
